File: Blog.Core/IEntity.cs
namespace Blog.Core
{
    public interface IEntity
    {

    }

    public interface IEntityBase : IEntity
    {
    }


    public interface IEntity<T> : IEntity
    {
        T Id { get; set; }
    }

    public interface IEntityBase<T> : IEntityBase
    {
        T Id { get; set; }
    }
}

File: Shared/Blog.FSharp.Core/IEntity.fs
namespace Blog.FSharp.Core

/// <summary>
/// Entity interface.
/// </summary>
type IEntity = interface end

/// <summary>
/// Entity base interface.
/// </summary>
type IEntityBase =
    inherit IEntity

/// <summary>
/// Entity interface.
/// </summary>
/// <typeparam name="T">Type.</typeparam>
type IEntity<'T> =
    inherit IEntity
    /// <summary>
    /// Gets or sets id.
    /// </summary>
    abstract member Id: 'T with get, set

/// <summary>
/// Entity base interface.
/// </summary>
/// <typeparam name="T">Type.</typeparam>
type IEntityBase<'T> =
    inherit IEntityBase
    inherit IEntity<'T>
